framework/src/modules/nft/method.ts

```diff
@@ -398,17 +360,11 @@ export class NFTMethod extends BaseMethod {
 			throw new Error('NFT does not exist');
 		}
 
-		const nft = await nftStore.get(methodContext, nftID);
-
-		if (nft.owner.length === LENGTH_CHAIN_ID) {
+		if (this.isNFTEscrowed(nft)) {
 			throw new Error('NFT is escrowed to another chain');
 		}
 
-		const userStore = this.stores.get(UserStore);
-		const userKey = userStore.getKey(nft.owner, nftID);
-		const userData = await userStore.get(methodContext, userKey);
-
-		if (userData.lockingModule === NFT_NOT_LOCKED) {
+		if (!this.isNFTLocked(nft)) {
```

Comment:
I just noticed out that in the `unlock` function the `LockEvent` has been used, I believe this should be corrected.

Reply:
True. Apparently the `UnlockEvent` is not used anywhere and does not even have a `error` method defined. I updated all affected places under [this](https://github.com/LiskHQ/lisk-sdk/pull/9005/commits/f7f5d0827c2cfac7dc6fca773f1cac041e9e0f8e) commit. Thanks for noticing this.